src/libraries/Microsoft.PowerFx.Core/Functions/IResourceManager.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    /// <summary>
+    /// Interface for resource manager.
+    /// </summary>
+    public interface IResourceManager
+    {
+        /// <summary>
+        /// Create a new resource element for string.
+        /// </summary>
+        /// <param name="str">String.</param>
+        /// <param name="fileType">Associated FileType.</param>
+        /// <returns>Index of the new element.</returns>
+        int NewElementFromString(string str, FileType fileType = FileType.Any);
+
+        /// <summary>
+        /// Create a new resource element for a base64 encoded string.
+        /// </summary>
+        /// <param name="base64str">Base64 encoded string.</param>
+        /// <param name="fileType">Associated FileType.</param>
+        /// <returns>Index of the new element.</returns>
+        int NewElementFromBase64String(string base64str, FileType fileType = FileType.Any);
+
+        /// <summary>
+        /// Get resource element for index.
```

Comment:
>// Get resource element for index. [](http://example.com/codeflow?start=9&length=34)

If we allow retrieval of an element by index, should we have an enumeration, or a Count property?

Reply:
Do we need to have it in the interface?
Could be left to the implementer